lib/filequeue.rb

```diff
@@ -52,25 +53,21 @@ def clear
 
   def safe_open(mode)
     File.open(@file_name, mode) do |file|
-      lock file
+   
+     lock file do
       yield file
+     end
+    
     end
   end
 
   # Locks the queue file for exclusive access.
   #
-  # Raises `FileLockError` if unable to acquire a lock.
-  #
-  # Return is undefined.
+  # Raises `Filelock::WaitTimeout` if unable to acquire a lock.
+  # Default timeout is 1 day
```

Comment:
That time is too long. 1 second would be better. If that seems to short we could make it configurable.

Reply:
Also it looks, from the file lock docs, that the default time out is 1 minute. That is probably ok.